CoreScriptsRoot/ServerCoreScripts/ClientChat/ChannelsBar.lua

```diff
@@ -281,6 +283,12 @@ function methods:CreateTweeners()
 	
 end
 
+function methods:WaitUntilParentedCorrectly()
```

Comment:
The way this new chat system has to do persist through death is kind of hacky where it parents everything to nil, then waits for the character to load, then parents everything back to PlayerGui. Because of this, there is at minimum a single frame where there could be messages that try to get put into the message log or channel tabs that try to be created. When they try to do size related stuff when they are not descendants of PlayerGui they will always return a size of 0. This causes messages to get bunched together and overlap. It also breaks the channel bar because it tries to do tweening to get all the tabs ordered and it will throw an error if it's not parented correctly.